pingcap/tidb
# br: get bucket region from s3 to enable users not to input s3 region param

### What problem does this PR solve?

Issue Number: close #34275

Problem Summary:

When users use S3 in br/dumpling/lightning, they must set both `region` and `bucket` to access the correct bucket. Actually, uses don't need to input `region`, because we can get `region` by bucket from S3. So we add this PR to enable users not to input `region`.

### What is changed and how it works?

We invoke `s3manager.GetBucketRegionWithClient`, which encapsulate the function of getting region by bucket from S3. Then we compare the region got from S3 and the region input, if they are different, we will correct region using region from S3.

### Check List

Tests 

       1. create a bucket in S3.
       2. add test code to invoke new function(adding `s3manager.GetBucketRegionWithClient`), then invoke `ReadFile` and    `WriteFile` to verify if it is valid.
       3. run the test code and check the result. test code and result are as follow:

![s3_test_code](https://user-images.githubusercontent.com/13897016/167355875-89159eea-c77c-469f-a530-ba945f26dee7.png)

![s3_test_result](https://user-images.githubusercontent.com/13897016/167096174-7c692700-0c11-47f6-9200-cd8b28302695.png)

Documentation


use can run br/dumpling/lighting without `region` param.

### Release note



Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Get bucket region from S3 automatically to enable users not to input region param when use S3.
```
